### PR TITLE
[TR] Bunch of fixes related to PEF

### PR DESCRIPTION
**Pull request by grena**
| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | Y
| Specs & Static     | Y
| Changelog updated  | N

### CODE REVIEW

**Comment by sumbobyboys** [comment]
such bien, much mortel. GTM

File: features/Context/Page/Product/Edit.php [hunk 1]
```diff
@@ -1415,7 +1420,15 @@ public function startCopy()
...
     public function compareWith($localeCode, $scope = null, $source = null)
     {
-        $this->startCopy();
+        try {
+            $this->startCopy();
+        } catch (\Exception $e) {
+            // Is panel already open?
+            $this->spin(function () {
+                return $this->getElement('Copy actions')->find('css', '.stop-copying');
+            }, 20, "Copy panel seems not open nor closed.");
```
Comment (phaseinducer) [inline features/Context/Page/Product/Edit.php]: "neither...nor..." ; )

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php [hunk 2]
```diff
@@ -340,14 +340,29 @@ protected function transformViolations(ConstraintViolationListInterface $violati
...
-                    'attribute'     => $value->getAttribute()->getCode(),
+                $currentError = [
+                    'attribute'     => $attributeCode,
                     'locale'        => $value->getLocale(),
                     'scope'         => $value->getScope(),
                     'message'       => $violation->getMessage(),
                     'invalid_value' => $violation->getInvalidValue()
                 ];
+
+                isset($errors['values'][$attributeCode]) || $errors['values'][$attributeCode] = [];
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php]: not very readable

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php [hunk 3]
```diff
@@ -340,14 +340,29 @@ protected function transformViolations(ConstraintViolationListInterface $violati
...
+
+                // Is error already in errors array?
+                $errorInArray = array_reduce(
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php]: maybe `array_filter()` or a simple foreach would be less classy but way more readable

File: src/Pim/Bundle/EnrichBundle/Normalizer/AttributeNormalizer.php [hunk 4]
```diff
@@ -48,13 +48,28 @@ public function __construct(
      */
     public function normalize($attribute, $format = null, array $context = [])
     {
+        $dateMin = (is_null($attribute->getDateMin())) ? '' : $attribute->getDateMin()->format(\DateTime::ISO8601);
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Normalizer/AttributeNormalizer.php]: `null ===`

File: src/Pim/Bundle/EnrichBundle/Normalizer/AttributeNormalizer.php [hunk 5]
```diff
@@ -48,13 +48,28 @@ public function __construct(
...
+        $dateMax = (is_null($attribute->getDateMax())) ? '' : $attribute->getDateMax()->format(\DateTime::ISO8601);
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Normalizer/AttributeNormalizer.php]: same

**Comment by phaseinducer** [comment]
:+1: 
GTM after little comments fix

**Pull request merged**